_posts/2019-03-18-rules-jvm-external-maven.md

```diff
@@ -0,0 +1,159 @@
+from community members:
+
+* [https://github.com/johnynek/bazel-deps](https://github.com/johnynek/bazel-deps)
+    * Tool to generate Bazel dependencies transitively for Maven artifacts, with
+      Scala and Kotlin support
+* [https://github.com/square/bazel\_maven\_repository](https://github.com/square/bazel_maven_repository)
+    * Repository rule to create an idiomatic Bazel representation of a Maven
+      repository using a pinned list of artifacts, with support for Kotlin and
+      Android dependencies
+* [https://github.com/menny/mabel](https://github.com/menny/mabel)
+    * Tool to generate a Maven dependency graph as a lock file, with support for
+      Kotlin and Android dependencies
+
+Each project approaches the problem slightly differently. Check out those
+projects as well to see their full feature set and determine which solution
+works best for you. It is also possible to wire dependencies up manually using
+rules like `java_import_external` and `aar_import_external`.
+
+## Objectives
+
+We wanted to fulfill two primary objectives with `rules_jvm_external`.
+
+The first is that users should be able to run `bazel build //...`. This means
+that the calculation of the transitive dependencies and their fetching need to
+happen during repository rule execution time. This makes dependency fetching
+transparent to users and reduces onboarding friction to a Bazel-built project.
+
+The second is to provide an interface that is familiar to users with projects
+that depend on Maven repositories. The specification of artifacts in the
+repository rule resembles the specification in the `<dependencies>` tag in a
+Maven POM file or the `dependencies` closure in a Gradle build file. In
+particular, this means that users only need to specify their direct dependencies
+and the rule will handle the rest for them.
+
+## Features and use cases
+
+For the initial release, we are targeting support for Java and Android Java
+projects. The following features are supported:
+
+* Supported on Windows, macOS and Linux
+* JAR and AAR packaging types
+* In-band transitive closure resolution
+* Dependency specification directly in the WORKSPACE file
+* Custom Maven repositories
+* Maven repository authentication with HTTP Basic Authentication
+* Artifact version pinning and resolution
+* Multiple independent closures of versioned artifacts
+    * Not forced to only have one version of an artifact per workspace, although
+      different versions are namespaced by the repository name
+* Source JAR support
+* Dependency exclusion
+
+As part of this, we will no longer be maintaining generate_workspace.
+gmaven_rules, which was designed to be an interim solution, will also be
+replaced by rules_jvm_external. We will be updating existing documentation to
+reflect this over the coming weeks.
+
+## How to use
+
+To start using this project in your Java or Android project, in your WORKSPACE
```

Comment:
Change to "...using this feature..."? Using a project in a project sounds like "butter is buttery".